daemon/daemon_test.go

```diff
@@ -77,19 +77,17 @@ type DaemonSuite struct {
 	OnClearPolicyConsumers    func(id uint16) *sync.WaitGroup
 }
 
-func (ds *DaemonSuite) SetUpTest(c *C) {
+func TestMain(m *testing.M) {
 	option.Config.Populate()
-	ds.oldPolicyEnabled = policy.GetPolicyEnabled()
-	policy.SetPolicyEnabled(option.DefaultEnforcement)
-
-	// kvstore is initialized before generic SetUpTest so it must have been completed
-	ds.kvstoreInit = true
-
 	time.Local = time.UTC
 	tempRunDir, err := ioutil.TempDir("", "cilium-test-run")
-	c.Assert(err, IsNil)
+	if err != nil {
+		panic("TempDir() failed.")
```

Comment:
You should be able to pass a third (or more) parameter to `c.Assert(err, IsNil)` that is treated as a loggable message. That would also include the error itself, unless you wanted to not print that?

Reply:
'TestMain()' does not have 'c', so no 'Assert' either. Figured that if these fail we could just panic instead.